addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/handler/SolarEdgeBaseHandler.java

```diff
@@ -0,0 +1,206 @@
+        }
+    }
+
+    /**
+     * Disposes the bridge.
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        if (liveDataPollingJob != null && !liveDataPollingJob.isCancelled()) {
+            logger.debug("stop live data polling job");
+            liveDataPollingJob.cancel(true);
+            liveDataPollingJob = null;
+        }
+        if (aggregateDataPollingJob != null && !aggregateDataPollingJob.isCancelled()) {
+            logger.debug("stop aggregate data polling job");
+            aggregateDataPollingJob.cancel(true);
+            aggregateDataPollingJob = null;
+        }
+    }
+
+    @Override
+    public @Nullable WebInterface getWebInterface() {
+        return webInterface;
+    }
+
+    /**
+     * will update all channels provided in the map
+     */
+    @Override
+    public void updateChannelStatus(Map<String, @Nullable String> values) {
+        logger.debug("Handling channel update.");
+
+        for (String key : values.keySet()) {
+            Channel channel = getThingSpecificChannel(key);
+            if (channel != null) {
+                String value = values.get(key);
+                logger.debug("Channel is to be updated: {}: {}", channel.getFQName(), value);
+                if (value != null) {
+                    if (channel.getJavaType().equals(Double.class)) {
+                        try {
+                            updateState(channel.getFQName(), convertToDecimal(value));
+                        } catch (NumberFormatException ex) {
+                            logger.warn("Could not update channel {} - invalid number: '{}'", channel.getFQName(),
+                                    value);
+                            updateState(channel.getFQName(), UnDefType.UNDEF);
+                        }
+                    } else {
+                        updateState(channel.getFQName(), new StringType(value));
+                    }
+                } else {
+                    logger.debug("Value is null or not provided by solaredge (channel: {})", channel.getFQName());
+                    updateState(channel.getFQName(), UnDefType.UNDEF);
+                }
+            } else {
+                logger.debug("Could not identify channel: {} for model {}", key,
+                        getThing().getThingTypeUID().getAsString());
+            }
+        }
+    }
+
+    /**
+     * internal method for conversion to a valid decimal value.
+     *
+     * @throws NumberFormatException
+     * @param number as String
+     * @return converted value to DecimalType
+     */
+    private DecimalType convertToDecimal(String number) {
+        return new DecimalType(number.replaceAll(",", ".").replaceAll("[^0-9.-]", ""));
+    }
+
+    @Nullable
+    protected abstract Channel getThingSpecificChannel(String id);
+
+    @Override
+    public void setStatusInfo(ThingStatus status, ThingStatusDetail statusDetail, String description) {
```

Comment:
This can be removed, can't it?

Reply:
The method "setStatusInfo" is used, so I cannot remove it.

Reply:
Ah, sorry, I only noticed now that it has a different name than `updateStatus`, which you call on the BaseThingHandler. I see that you need this method as you define it in your own interface. So all fine.